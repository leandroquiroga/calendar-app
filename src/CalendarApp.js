import React from 'react'

export const CalendarApp = () => {
  return (
    <h1>Hola Mundo ! </h1>
  )
}
