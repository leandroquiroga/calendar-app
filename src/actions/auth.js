import { fetchNotToken, fetchWithToken } from "../helpers/fetchAPI"
import { types } from "../types/types";

const login = (user) => ({
  type: types.authLogin,
  payload: user
});

const checkingFinish = () => ({ type: types.authChekingFinish });

const logout = () => ({ type: types.authLogout });

const eventLogout = () => ({ type: types.eventLogout });

export const authStartLogin = (email, password, setMsgError) => {
  return async (dispatch) => {
    const response = await fetchNotToken('auth/login', { email, password }, 'POST');
    const body = await response.json();

    // En caso de que haya un error en el login retorna el mensaje de error.
    if (!body.ok) return setMsgError(body.msg);

    localStorage.setItem('token', body.token);
    localStorage.setItem('token-init-date', new Date().getTime());
    dispatch(login({
      uid: body.uid,
      name: body.name
    }));
  }
} 

export const startRegister = (email, password, name, setMsgError) => {
  return async (dispatch) => {
    const response = await fetchNotToken('auth/register', { email, password, name }, 'POST');
    const body = await response.json();

    // En caso de que haya un error en el registro retornara un mensaje de error 
    if (!body.ok) return setMsgError(body.msg);

    localStorage.setItem('token', body.token);
    localStorage.setItem('token-init-date', new Date().getTime());
    dispatch(login({
      uid: body.uid,
      name: body.name,
    }));
  };
}


export const startCheking = () => {
  return async (dispatch) => {
    const response = await fetchWithToken('auth/renew');
    const body = await response.json();

    if (!body.ok) {
      // Cambia el checking a true y retorna un mensaje de error desde el backend
      return dispatch(checkingFinish());
    };

    localStorage.setItem('token', body.token);
    localStorage.setItem('token-init-date', new Date().getTime());

    dispatch(login({
      uid: body.uid,
      name: body.name,
    }));
  };
};


export const startUserLogut = () => {
  return (dispatch) => {
    // Debe eliminar el token
    localStorage.removeItem('token');
    localStorage.removeItem('token-init-date');

    dispatch(eventLogout())
    dispatch(logout());
  }
};