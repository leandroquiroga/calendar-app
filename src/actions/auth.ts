import { Dispatch, SetStateAction } from "react";

import { AppDispatch } from "../store/store";
import { fetchNotToken, fetchWithToken } from "../helpers/fetchAPI"
import { types } from "../types/types";

export interface UserProps {
  email?: string;
  password?: string;
  name: string;
  uid: string;
};

export interface UserLogin {
  email: string;
  password: string;
  setMsgError: Dispatch<SetStateAction<string>>
};

export interface UserRegister {
  email: string;
  password: string;
  name?: string;
  setMsgError: Dispatch<SetStateAction<string>>
};

const login = (user: UserProps) => ({
  type: types.authLogin,
  payload: user
});

const checkingFinish = () => ({ type: types.authChekingFinish });

const logout = () => ({ type: types.authLogout });

const eventLogout = () => ({ type: types.eventLogout });

export const authStartLogin = (
  email: string,
  password: string,
  setMsgError: Dispatch<SetStateAction<string>>
): any => {
  return async (dispatch: AppDispatch) => {
    const response = await fetchNotToken(
      "auth/login",
      { email, password },
      "POST"
    );
    const body = await response.data;

    // En caso de que haya un error en el login retorna el mensaje de error.
    if (!body.ok) return setMsgError(body.msg);

    localStorage.setItem("token", body.token);
    localStorage.setItem("token-init-date", new Date().getTime().toString());
    dispatch(
      login({
        uid: body.uid,
        name: body.name,
      })
    );
  };
}; 

export const startRegister = (
  email: string,
  password: string,
  name: string,
  setMsgError: Dispatch<SetStateAction<string>>,
): any => {
  return async (dispatch: AppDispatch) => {
    const response = await fetchNotToken(
      "auth/register",
      { email, password, name },
      "POST"
    );
    const body = await response.data;

    // En caso de que haya un error en el registro retornara un mensaje de error
    if (!body.ok) return setMsgError(body.msg);

    localStorage.setItem("token", body.token);
    localStorage.setItem("token-init-date", new Date().getTime().toString());
    dispatch(
      login({
        uid: body.uid,
        name: body.name,
      })
    );
  };
};

// Buscar una mejor definicion para el tipo de retorno de la funcion
export const startCheking = (): any => {
  return async (dispatch: AppDispatch) => {
    const response = await fetchWithToken("auth/renew", "GET");
    const body = await response.data;
    if (!body.ok) {
      // Cambia el checking a true y retorna un mensaje de error desde el backend
      return dispatch(checkingFinish());
    }

    localStorage.setItem("token", body.token);
    localStorage.setItem("token-init-date", new Date().getTime().toString());

    dispatch(
      login({
        uid: body.uid,
        name: body.name,
      })
    );
  };
};

// Buscar una mejor definicion para el tipo de retorno de la funcion
export const startUserLogut = ():any => {
  return (dispatch: AppDispatch) => {
    // Debe eliminar el token
    localStorage.removeItem("token");
    localStorage.removeItem("token-init-date");

    dispatch(eventLogout());
    dispatch(logout());
  };
};