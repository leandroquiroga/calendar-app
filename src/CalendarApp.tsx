import React, { useState } from 'react'
import moment from 'moment';
import { Calendar, momentLocalizer, View } from 'react-big-calendar';
import { useDispatch, useSelector } from 'react-redux';

import { uiOpenModal } from './actions/actions';
import { clearEventActive, eventSetActive } from './actions/events';
import { RootState } from './store/store';
import { messages_es } from './helpers/calendar-messages-es';

import { Navbar } from './components/ui/Navbar'
import { AddNewFab } from './components/ui/AddNewFab';
import { DeleteEventFab } from './components/ui/DeleteEventFab';
import { CalendarEvent } from './components/calendar/CalendarEvent';
import { CalendarModal } from './components/calendar/CalendarModal';

import 'moment/locale/es';
import 'react-big-calendar/lib/css/react-big-calendar.css';

// Configuracion en espaÃ±ol de moment
moment.locale('es') 

// Configuracion del localizer
const localizer = momentLocalizer(moment);

export const CalendarApp = () => {

  const dispatch = useDispatch();
  const { events, activeEvent } = useSelector((state: RootState) => state.calendar);
  const { uid } = useSelector((state: RootState) => state.auth);

  // const { user } = activeEvent;

  // Mantiene el estado de la ultima vista
  const [lastView, setLastView] = useState<any>(
    localStorage.getItem("lastView") || "month"
  );
  
  // Dispara la accion para abrir el modal
  const handleOnDoubleClick = () => dispatch(uiOpenModal());

  // Evento seleccionado
  const handleEventSelect = (e: any) => dispatch(eventSetActive(e));
  
  // Evento de cambio de vista
  const handleOnViewChange = (e: View) => {
    setLastView(e);
    localStorage.setItem("lastView", e);
  };

  const onSelectSlot = () => {
    dispatch(clearEventActive());
  };

  const eventSytleGetter = (event: any) => {
    const { user } = event;

    const style = {
      backgroundColor: (uid === user._id) ? '#1b61db' : '#202020',
      borderRadius: '0px',
      color: 'white',
      opacity: 0.9,
      outlineColor: '#bcbcbc',
    };

    return {
      style,
    }
  }
  return (
    <section className='calendar_container'>
      <Navbar />
      <Calendar
        popup={true}
        localizer={localizer}
        events={events}
        messages={messages_es}
        eventPropGetter={eventSytleGetter}
        startAccessor='start'
        endAccessor='end'
        step={7.5}
        onSelectEvent={handleEventSelect}
        onSelectSlot={onSelectSlot}
        selectable={true}
        onView={handleOnViewChange}
        view={lastView}
        onDoubleClickEvent={handleOnDoubleClick}
        components={{
          event: CalendarEvent,
        }}
      />

      <CalendarModal />

      <AddNewFab />

      {(activeEvent)
        ? (activeEvent.user._id === uid) && <DeleteEventFab />
        : null
      }

    </section>
  )
};