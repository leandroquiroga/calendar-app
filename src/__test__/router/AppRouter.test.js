import React from 'react';
import { mount} from "enzyme";
import { Provider } from 'react-redux';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

import { storeAppRouter } from "../../utilities/utilities";
import { AppRourter } from "../../router/AppRourter";
import { PublicRoute } from '../../router/PublicRoute';
import { Register } from '../../components/auth/Register';
import { Login } from '../../components/auth/Login';
import { PageNotFound } from '../../components/pageNotFound/PageNotFound';
import { PrivateRoute } from '../../router/PrivateRoute';
import { CalendarApp } from '../../CalendarApp';

import '@testing-library/jest-dom';
import '../../setupTest';

jest.mock('../../actions/auth', () => ({
  startCheking: jest.fn()
}));


storeAppRouter.dispatch = jest.fn();


describe('Test on <AppRouter />', () => {

  test('this test should make a snapshot with component <Loading /> corretly', () => {

    const wapper = mount(
      <Provider store={storeAppRouter}>
        <AppRourter />
      </Provider>
    );

    expect(wapper).toMatchSnapshot();
  });

  test('this test should make a snapshot with component <Login /> corretly', () => {

    const stateChecking = storeAppRouter.getState();

    stateChecking.auth.checking = false

    const wapper = mount(
      <Provider store={storeAppRouter}>
        <MemoryRouter initialEntries={['/login']}>
           <Routes>
            <Route path='login' element={
              <PublicRoute>
                <Login />
              </PublicRoute>
              }/>
           </Routes>
         </MemoryRouter>
      </Provider>
    );

    expect(wapper).toMatchSnapshot();
    expect(wapper.find('.btn-primary').text().trim()).toBe('Iniciar sesi√≥n');

  });

   test('this test should make a snapshot with comoponent <Register /> correctle', () => {

     const wapper = mount(
       <Provider store={storeAppRouter}>
         <MemoryRouter initialEntries={['/register']}>
           <Routes>
            <Route path='register' element={
              <PublicRoute>
                <Register />
              </PublicRoute>
              }/>
           </Routes>
         </MemoryRouter>
      </Provider>
    );
    expect(wapper).toMatchSnapshot()
    expect(wapper.find('.btn-primary').text().trim()).toBe('Registrarme');

   });

  test('this test should make a snapshot with component <PageNotFound /> correctly', () => {

    const wapper = mount(
      <Provider store={storeAppRouter}>
        <MemoryRouter initialEntries={['/*']}>
          <Routes>
            <Route path='/*' element={<PageNotFound />} />
          </Routes>
        </MemoryRouter>
      </Provider>
    );

    expect(wapper).toMatchSnapshot();
    expect(wapper.find('.btn-dark').text().trim()).toBe('Regresar');
  });

  test('this test should show the children of the component <PrivateRoute />', () => {

    const uidFake = storeAppRouter.getState().auth;
    const calendarFake = storeAppRouter.getState().calendar;

    uidFake.uid = 'Testing123';

    calendarFake.event = [{
      title:"Backend Digital",
      notes:"",
      start:"2022-06-02T12:45:00.000Z",
      end: "2022-06-02T13:00:00.000Z",
      user: {
        _id:"6285455405b1fddeb0b5a999",
        name:"Leandro"

      },
      id:"62978ddd74ecd43d8c87ea48",
    }];

    const wapper = mount(
      <Provider store={storeAppRouter}>
        <MemoryRouter initialEntries={['/']}>
          <Routes>
            <Route path='/' element={
              <PrivateRoute>
                <CalendarApp />
              </PrivateRoute>
            } />
          </Routes>
        </MemoryRouter>
      </Provider>
    );

    expect(wapper).toMatchSnapshot();
  });
});