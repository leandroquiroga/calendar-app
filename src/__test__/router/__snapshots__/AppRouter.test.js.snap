// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter /> this test should make a snapshot with comoponent <Register /> correctle 1`] = `
<section
  className="row p-0 m-0"
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <article
    className="col-md-6 d-flex flex-column justify-content-center align-items-center"
  >
    <div
      className="border rounded-circle overflow-hidden shadow-sm p-3"
    >
      <img
        alt="register"
        className="img-fluid"
        src="register.svg"
        style={
          Object {
            "height": "250px",
            "width": "250px",
          }
        }
      />
    </div>
    <div
      className="text-center p-2 mt-3"
    >
      <p
        className="fs-5 fw-bold"
      >
         !Bienvenido a Calendar! 
      </p>
      <small
        className="fw-normal fs-6"
      >
         Registrate para crear tu agenda web. 
      </small>
      <hr />
      <span
        className="m-0"
      >
        ¿Tienes cuenta? Haz click en 
        <a
          className="text-decoration-none"
          href="/login"
          onClick={[Function]}
        >
          Inicia sesión
        </a>
      </span>
    </div>
  </article>
  <article
    className="col-md-6 bg-primary d-flex flex-column justify-content-center align-items-center"
  >
    <h1
      className="text-white fs-3 p-4"
    >
       Registrarte 
    </h1>
    <form
      className="form-control my-5 form_container bg-white border-0 rounded p-3"
      onSubmit={[Function]}
    >
      <div
        className="mb-3 d-flex flex-column"
      >
        <label
          className="fw-normal form-label"
          htmlFor="name"
        >
          Nombre
        </label>
        <input
          className="border-0 border-bottom form_container__field"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Ingrese su nombre..."
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-3 d-flex flex-column"
      >
        <label
          className="fw-normal form-label"
          htmlFor="email"
        >
          Correo electronico:
        </label>
        <input
          className="border-0 border-bottom form_container__field"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Ingrese su email..."
          type="email"
          value=""
        />
      </div>
      <div
        className="mb-3 d-flex flex-column"
      >
        <label
          className="fw-normal form-label"
          htmlFor="password"
        >
          Contraseña:
        </label>
        <input
          className="border-0 border-bottom form_container__field"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Ingrese su contraseña..."
          type="password"
          value=""
        />
      </div>
      <div
        className="mb-3 d-flex flex-column"
      >
        <label
          className="fw-normal form-label"
          htmlFor="passwordConfirm"
        >
          Repetir contraseña:
        </label>
        <input
          className="border-0 border-bottom form_container__field"
          id="passwordConfirm"
          name="passwordConfirm"
          onChange={[Function]}
          placeholder="Repita su contraseña..."
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-primary rounded p-2 form_container__button form_container__button-disabled"
        disabled={true}
        type="submit"
      >
        Registrarme
      </button>
    </form>
  </article>
</section>
`;

exports[`Test on <AppRouter /> this test should make a snapshot with component <Loading /> corretly 1`] = `
<section
  className="d-flex justify-content-center align-items-center min-vh-100 bg-primary"
>
  <div
    className="lds-ring"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</section>
`;

exports[`Test on <AppRouter /> this test should make a snapshot with component <Login /> corretly 1`] = `
<section
  className="row p-0 m-0"
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <article
    className="col-md-6 d-flex flex-column justify-content-center align-items-center"
  >
    <div
      className="border rounded-circle overflow-hidden shadow-sm p-3"
    >
      <img
        alt="login"
        className="img-fluid"
        src="login.svg"
        style={
          Object {
            "width": "250px",
          }
        }
      />
    </div>
    <div
      className="text-center p-2 mt-3"
    >
      <p
        className="fs-5 fw-bold"
      >
         !Bienvenido a Calendar! 
      </p>
      <small
        className="fw-normal fs-6"
      >
         Inicie sesión para abrir su agenda web. 
      </small>
      <hr />
      <span
        className="m-0"
      >
        ¿No tienes cuenta? Haz click en 
        <a
          className="text-decoration-none"
          href="/register"
          onClick={[Function]}
        >
          Registrarme
        </a>
      </span>
    </div>
  </article>
  <article
    className="col-md-6 bg-primary d-flex flex-column justify-content-center align-items-center"
  >
    <h1
      className="text-white fs-3 p-4"
    >
       Inicia Sesión 
    </h1>
    <form
      className="form-control my-5 form_container bg-white border-0 rounded p-3"
      onSubmit={[Function]}
    >
      <div
        className="mb-3 d-flex flex-column"
      >
        <label
          className="fw-normal form-label"
          htmlFor="email"
        >
          Correo electronico:
        </label>
        <input
          className="border-0 border-bottom form_container__field"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Ingrese su email..."
          type="email"
          value=""
        />
      </div>
      <div
        className="mb-3 d-flex flex-column"
      >
        <label
          className="fw-normal form-label"
          htmlFor="password"
        >
          Contraseña:
        </label>
        <input
          className="border-0 border-bottom form_container__field"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Ingrese su contraseña..."
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-primary rounded p-2 form_container__button form_container__button-disabled"
        disabled={true}
        type="submit"
      >
        Iniciar sesión
      </button>
    </form>
  </article>
</section>
`;

exports[`Test on <AppRouter /> this test should make a snapshot with component <PageNotFound /> correctly 1`] = `
<section
  className="d-flex flex-column justify-content-center align-items-center"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <article
    className="container p-2"
  >
    <img
      alt="logo 404"
      className="d-block mx-auto img-fluid"
      src="error404.svg"
      style={
        Object {
          "width": "350px",
        }
      }
    />
    <p
      className="text-center fs-3"
    >
      Lo sentimos, no se ha encontrado la URL
    </p>
  </article>
  <button
    className="btn btn-dark p-2"
    onClick={[Function]}
  >
    Regresar
  </button>
</section>
`;

exports[`Test on <AppRouter /> this test should show the children of the component <PrivateRoute /> 1`] = `
<section
  className="calendar_container"
>
  <nav
    className="navbar navbar-primary bg-primary"
  >
    <section
      className="container-fluid"
    >
      <span
        className="text-white text-capitalize"
      >
        Hola, 
         !
      </span>
      <button
        className="btn btn-outline-light p-2 rounded"
        onClick={[Function]}
      >
        <i
          className="fa-solid fa-arrow-right-from-bracket"
        />
        <span>
           Logout 
        </span>
      </button>
    </section>
  </nav>
  <div
    className="rbc-calendar"
  >
    <div
      className="rbc-toolbar"
    >
      <span
        className="rbc-btn-group"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Hoy
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Anterior
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Siguiente
        </button>
      </span>
      <span
        className="rbc-toolbar-label"
      >
        junio 2022
      </span>
      <span
        className="rbc-btn-group"
      >
        <button
          className="rbc-active"
          key="month"
          onClick={[Function]}
          type="button"
        >
          Mes
        </button>
        <button
          className=""
          key="week"
          onClick={[Function]}
          type="button"
        >
          Semana
        </button>
        <button
          className=""
          key="day"
          onClick={[Function]}
          type="button"
        >
          Día
        </button>
        <button
          className=""
          key="agenda"
          onClick={[Function]}
          type="button"
        >
          Agenda
        </button>
      </span>
    </div>
    <div
      aria-label="Month View"
      className="rbc-month-view"
      role="table"
    >
      <div
        className="rbc-row rbc-month-header"
        role="row"
      >
        <div
          className="rbc-header"
          key="header_0"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            lun.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_1"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mar.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_2"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mié.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_3"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            jue.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_4"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            vie.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_5"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            sáb.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_6"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            dom.
          </span>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_0"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                30
              </button>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_1"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                31
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                01
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                02
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                03
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                04
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                05
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-today"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                06
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                07
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                08
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                09
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                10
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                11
              </button>
            </div>
            <div
              className="rbc-date-cell rbc-now rbc-current"
              key="header_6"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                12
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                13
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                14
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                15
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                16
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                17
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                18
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                19
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                20
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                21
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                22
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                23
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                24
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                25
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                26
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                27
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                28
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                29
              </button>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                30
              </button>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_4"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                01
              </button>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_5"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                02
              </button>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_6"
              role="cell"
            >
              <button
                className="rbc-button-link"
                onClick={[Function]}
                role="cell"
                type="button"
              >
                03
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <article
    className="d-flex justify-content-center"
  >
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        />
      }
    />
  </article>
  <button
    className="btn btn-primary shadow rounded-circle button_icon"
    onClick={[Function]}
  >
    <i
      className="fa-solid fa-plus"
    />
  </button>
</section>
`;
